File: Nickvision.Application.Shared/Events/FolderChangedEventArgs.cs
using System;
using System.Collections.Generic;

namespace Nickvision.Application.Shared.Events;

public class FolderChangedEventArgs : EventArgs
{
    public string? Path { get; init; }
    public List<string> Files { get; init; }

    public bool IsOpen => !string.IsNullOrEmpty(Path);

    public FolderChangedEventArgs(string? path, List<string>? files = default)
    {
        Path = path;
        Files = files ?? [];
    }
}

File: libapplication/src/events/folder_changed_event_args.cpp
#include "events/folder_changed_event_args.h"

namespace application::events
{
	folder_changed_event_args::folder_changed_event_args(std::filesystem::path path, std::vector<std::filesystem::path> files)
	    : m_path{ std::move(path) },
	      m_files{ std::move(files) }
	{
	}

	const std::filesystem::path& folder_changed_event_args::get_path() const
	{
		return m_path;
	}

	const std::vector<std::filesystem::path>& folder_changed_event_args::get_files() const
	{
		return m_files;
	}
}